packages/golden-test-generator/src/ChainSync/chainSync.ts

```diff
@@ -0,0 +1,126 @@
+import { GeneratorMetadata } from '../Content';
+import { Logger } from 'ts-log';
+import { Ogmios } from '@cardano-sdk/ogmios';
+
+export type RollForward = { type: 'rollForward'; block: Ogmios.Schema.Block };
+export type RollBackward = { type: 'rollBackward'; rollback: Ogmios.Schema.TipOrOrigin };
+export type ChainSyncEvent = RollForward | RollBackward;
+
+export type GetBlocksResponse = GeneratorMetadata & {
+  events: ChainSyncEvent[];
+};
+
+type RequestedBlocks = { [blockHeight: number]: Ogmios.Schema.Block };
+
+const getBlockHeaderAndHash = (block: Ogmios.Schema.Block) => {
```

Comment:
You could use `ogmiosToCore.getBlock` utility to get the required info without having to handle specific block types:
```diff
+ const {header: {blockNo: blockHeight, hash, slot}} = ogmiosToCore.getBlock(block);

Reply:
Took it a step further and updated `golden-test-generator` chain-sync to write blocks in `core` format instead of Ogmios:
e6932f013ca5a1b98b4c3b60a6a10145f51c22d5
8a26d22ba796cf8302fd34109b13306664331848
580e09e8e1c257d184c8afdab85fe82315145746
5e5881e9aa505a45d5f6d58c5296b7240755dc55
4524a62a7dcdf12ea2e5b5b119a1b3662ffaf350

Reply:
580e09e8e1c257d184c8afdab85fe82315145746 should be included with the changes in 4524a62a7dcdf12ea2e5b5b119a1b3662ffaf350. Might need to do some squashing to have clarity over the atomic changes and the appropriate sequence